[lint] Add Verilator UNOPTFLAT lint waivers

alees24 opened the pull request:
These lint waivers are required in the Sonata FPGA project when the CHERIoT Ibex core is connected directly through the TL-UL xbar to tlul_sram_adapter modules. They prevent spurious reports of circular combinational logic (UNOPTFLAT); in this case Verilator is warning of the impact of additional delta cycles upon simulation, not combinational loops in the logic.

marnovandermaas commented on vendor/lowrisc_ip/ip/prim/lint/prim_arbiter.vlt:
This should be put in a patch file I think.

alees24 replied:
Yes, that was the intent, thanks. Take two.

alees24 committed:
Patch to add UNOPTFLAT waiver to prim_arbiter_ppc

This waiver is required for Sonata FPGA builds when the CHERIoT Ibex
core is connected directly through the xbar to tlul_adapter_sram.

alees24 committed:
Patched prim_arbiter.vlt file

Add the patched file from the patch in the previous commit;
UNOPTFLAT waiver for Sonata FPGA builds to prevent spurious
reports of combinational loops.

alees24 committed:
Add UNOPTFLAT waiver to ibex_id_stage

When the core is connected directly through the xbar/TL-UL
fabric to tlul_sram_adapter with no register delays, prevent
Verilator UNOPTFLAT warnings breaking simulations of the Sonata
system; this is not a loop in combinational logic, just a warning
about reduced simulation performance.

marnovandermaas reviewed (approved):
Looks good to me. I'll merge this so you can vendor it in on the Sonata side.
